backend/src/main/kotlin/app/ehrenamtskarte/backend/auth/service/Authorizer.kt

```diff
@@ -80,30 +60,19 @@ object Authorizer {
         newAdminRole: Role,
         newAdminRegion: RegionEntity?
     ): Boolean {
-        if (actingAdmin.projectId.value != newAdminProjectId) {
-            return false
-        } else if (newAdminRole == Role.NO_RIGHTS) {
+        if (actingAdmin.projectId.value != newAdminProjectId || newAdminRole == Role.NO_RIGHTS) {
             return false
         }
-
-        if (actingAdmin.role == Role.PROJECT_ADMIN.db_value && newAdminRole == Role.EXTERNAL_VERIFIED_API_USER) {
-            return transaction {
-                ProjectEntity.find { Projects.project eq EAK_BAYERN_PROJECT }
-                    .single().id.value == actingAdmin.projectId.value
+        if (actingAdmin.isRole(Role.PROJECT_ADMIN)) {
+            return newAdminRole != Role.EXTERNAL_VERIFIED_API_USER || transaction {
+                actingAdmin.isProject(
+                    EAK_BAYERN_PROJECT
+                )
             }
```

Comment:
I feel like negating statements, using disjunctions, and packing everything into a return statement makes it harder to read (at least for me).

```suggestion
            if (newAdminRole == Role.EXTERNAL_VERIFIED_API_USER && transaction { actingAdmin.isProject(EAK_BAYERN_PROJECT) }) {
              // Currently, we only allow external API users for the EAK Bayern project.
              return false
            }
            return true
```

Reply:
Agreed, negating should be avoided as possible. Will adjust accordingly.